PythonToolbox/quantconnect/api.py

```diff
@@ -11,448 +11,497 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from datetime import datetime
-from json import dumps
-from requests import get as requests_get, Request
-from time import mktime
-from quantconnect import ApiConnection
+from base64 import b64encode
+from datetime import datetime as dt
+from hashlib import sha256
+from json import dumps, loads
+from requests import get, post
+from time import mktime, time
+from quantconnect.Result import Result
 
 DOWNLOAD_CHUNK_SIZE = 256 * 1024
 
 class Api:
-    """QuantConnect.com Interaction Via API.
-    Attributes:
-        userId(int/str): User Id number from QuantConnect.com account. Found at www.quantconnect.com/account.
-        token(str): Access token for the QuantConnect account. Found at www.quantconnect.com/account.
-    """
-    def __init__(self, userId, token):
-        self.api_connection = ApiConnection(userId, token)
-
-    def connected(self):
-        """Check if Api is successfully connected with correct credentials"""
-        return self.api_connection.connected()
-
-    def read_project(self, projectId):
-        """Get details about a single project.
+    '''QuantConnect.com Interaction Via API.
```

Comment:
Wondering why we need a python implementation of the API, couldn't we use PythonNet to use existing API implementation?

Reply:
We want to provide a 100% Python implementation to avoid pythonnet compatibility issues with the user python version.